README.md
# UA-Sample-App
A basic implementation of the UA Library


CHANGELOG
Android ChangeLog
=================

Migration Guide: http://docs.urbanairship.com/topic_guides/android_migration.html.

Version 5.0.1 - October 8, 2014
===============================
 - Fix issue with channel not creating when missing Google Play Services for GCM registration.
 - Fix issue with the Urban Airship Javascript interface not loading for custom web views that extend
   UAWebView.

Version 5.0.0 - October 1, 2014
===============================

New Features
------------
 - Unified support for both GCM (Google) and ADM (Amazon) transports.
 - Includes support for interactive notifications. Includes more than 25 built-in
   interactive notification sets, including button resources for 9 languages.
   Additional/replacement localization strings may be added to built-in actions.
 - Includes a new action for social sharing that can be called from pushes or web views.
 - Includes support for defining custom events in our reporting system.
 - Rewritten location module with a simplified API and support for Google's Fused
   location provider.
 - Support for asynchronous takeoff and access to the Urban Airship Library to avoid
   blocking the main thread.
 - Introduced a new device identifier, the Channel ID, which will replace APIDs as the push address
   in Urban Airship API calls.
 - Includes support for registering multiple GCM sender Ids. Only the main GCM sender
   ID's messages will be handled by Urban Airship. The rest will be ignored.
 - Simplified push integration. The push intent receiver is now optional and the library
   supports automatically launching the application if the push intent receiver is not
   set or does not handle the open notification broadcasts.
 - A new default notification factory that uses the Big Text notification style.
 - User notifications can be disabled without disabling push. This allows the application
   to still be messaged when notifications are disabled.
 - The push broadcasts for push received, push opened, and registration events have
   been updated. A base broadcast receiver 'BaseIntentReceiver' can be extended
   to parse the intents and provide convenient listener style callbacks.
 - It is no longer necessary to check for Urban Airship Actions when deciding to
   launch the application in the push intent receiver.

Packaging Changes
-----------------
 - Urban Airship library is now a library project with resources. Eclipse users will
   have to import the library as a project and reference it in the projects settings. An
   aar package is provided for Android Studio users.
 - The library now depends on the latest version of the v4 support library.
 - GCM and Fused location depends on the Google Play Services library.

Sample Changes
--------------
 - Samples now set up to work with both GCM (Google) and ADM (Amazon).


Amazon Only - Version 4.0.0 - July 14, 2014
===========================================
 - Initial Amazon release.

Version 4.0.4 - August 5, 2014
==============================
 - Fix regression in 4.0.3 that broke webview actions for devices running API 17 or higher.

Version 4.0.3 - July 21, 2014
=============================
 - Add if-modified-since header and accept 304 not modified responses for Rich Application message listing.
 - Remove any usage of addJavascriptInterface for Android devices older than API 17 to prevent abuse of upstream security issue CVE-2012-6636.

Samples Changes:
 - Updated to use the latest Android Gradle plugin and build tools.

Version 4.0.2 - April 30, 2014
==============================
 - Fix NPE when defining a Landing Page Activity with no meta data elements.
 - Fix location security issue where all location data was broadcasted with
   implicit intents. Thank you to Yu-Cheng Lin for reporting this issue.
 - Added ability to set the intent receiver for location updates on UALocationManager.

Push Sample Changes:
 - Use local broadcast receiver from the support package to broadcast intents to
 the application.

Version 4.0.1 - March 26, 2014
==============================
- Fix possible rich push user token corruption.

Version 4.0.0 - March 25, 2014
==============================
- Added Urban Airship Actions framework - a generic framework that provides a convenient way to
automatically perform tasks by name in response to push notifications, Rich App Page interactions and JavaScript.
- Added UAWebViewClient class to be used with RichPushMessageWebView and LandingPageWebView to
provide proper auth and inject the javascript bridge.
- Renamed RichPushMessageView to RichPushMessageWebView.
- Removed ability to set a custom Rich Push Javascript Interface and namespace through RichPushManager.
- Removed RichPushMessageJavascriptInteface and RichPushMesssageJavascript in favor of the new UAJavascriptInterface.

- Deprecated "urbanairship" Javascript namespace.  iOS and Android now both use the common namespace "UAirship".
- Deprecated Javascript interface methods isMessageRead, markMessageRead, markMessageUnread,
navigateTo, getViewWidth, getViewHeight, and getDeviceOrientation.

Rich Push Sample Changes:
 - Updated AndroidManifest.xml for Actions.
 - Added custom Landing Page layout and styles.
 - Added MessagePagerFragment to display Rich Application Pages in a view pager.
 - Refactored inbox sample code to move most of the logic to fragments.
 - Added deep linking.

Push Sample Changes:
 - Updated AndroidManifest.xml for Actions.
 - Added deep linking.

Version 3.3.2 - March 7, 2014 - Internal Release
================================================
- Use nondestructive User api requests.  Allows multiple device tokens and APIDS
to share a single user.
- Deprecated RichPushUser getApids, setApids, addApids.

Version 3.3.1 - February 24, 2014
=================================
- Persist Urban Airship push ids to help prevent duplicate pushes.
- Fix analytics crash when using autopilot with background location.

Version 3.3.0 - January 15, 2014
================================

javadocs/assets/navtree_data.js
var NAVTREE_DATA =
[ [ "com.urbanairship", "com/urbanairship/package-summary.html", [ [ "Interfaces", null, [ [ "Cancelable", "com/urbanairship/Cancelable.html", null, "" ], [ "PendingResult", "com/urbanairship/PendingResult.html", null, "" ], [ "PendingResult.ResultCallback", "com/urbanairship/PendingResult.ResultCallback.html", null, "" ], [ "UAirship.OnReadyCallback", "com/urbanairship/UAirship.OnReadyCallback.html", null, "" ] ]
, "" ], [ "Classes", null, [ [ "AirshipConfigOptions", "com/urbanairship/AirshipConfigOptions.html", null, "" ], [ "ApplicationMetrics", "com/urbanairship/ApplicationMetrics.html", null, "" ], [ "Autopilot", "com/urbanairship/Autopilot.html", null, "" ], [ "BaseManager", "com/urbanairship/BaseManager.html", null, "" ], [ "CoreActivity", "com/urbanairship/CoreActivity.html", null, "" ], [ "CoreReceiver", "com/urbanairship/CoreReceiver.html", null, "" ], [ "Logger", "com/urbanairship/Logger.html", null, "" ], [ "UAirship", "com/urbanairship/UAirship.html", null, "" ], [ "UrbanAirshipProvider", "com/urbanairship/UrbanAirshipProvider.html", null, "" ], [ "UrbanAirshipResolver", "com/urbanairship/UrbanAirshipResolver.html", null, "" ] ]
, "" ], [ "Annotations", null, [ [ "ConstantClass", "com/urbanairship/ConstantClass.html", null, "" ], [ "PropertyName", "com/urbanairship/PropertyName.html", null, "" ] ]
, "" ] ]
, "" ], [ "com.urbanairship.actions", "com/urbanairship/actions/package-summary.html", [ [ "Interfaces", null, [ [ "ActionCompletionCallback", "com/urbanairship/actions/ActionCompletionCallback.html", null, "" ] ]
, "" ], [ "Classes", null, [ [ "Action", "com/urbanairship/actions/Action.html", null, "" ], [ "Action.ActivityResult", "com/urbanairship/actions/Action.ActivityResult.html", null, "" ], [ "ActionActivity", "com/urbanairship/actions/ActionActivity.html", null, "" ], [ "ActionArguments", "com/urbanairship/actions/ActionArguments.html", null, "" ], [ "ActionArguments.Builder", "com/urbanairship/actions/ActionArguments.Builder.html", null, "" ], [ "ActionRegistry", "com/urbanairship/actions/ActionRegistry.html", null, "" ], [ "ActionRegistry.Entry", "com/urbanairship/actions/ActionRegistry.Entry.html", null, "" ], [ "ActionResult", "com/urbanairship/actions/ActionResult.html", null, "" ], [ "ActionRunner", "com/urbanairship/actions/ActionRunner.html", null, "" ], [ "ActionService", "com/urbanairship/actions/ActionService.html", null, "" ], [ "ActionUtils", "com/urbanairship/actions/ActionUtils.html", null, "" ], [ "AddCustomEventAction", "com/urbanairship/actions/AddCustomEventAction.html", null, "" ], [ "DeepLinkAction", "com/urbanairship/actions/DeepLinkAction.html", null, "" ], [ "LandingPageAction", "com/urbanairship/actions/LandingPageAction.html", null, "" ], [ "LandingPageActivity", "com/urbanairship/actions/LandingPageActivity.html", null, "" ], [ "OpenExternalUrlAction", "com/urbanairship/actions/OpenExternalUrlAction.html", null, "" ], [ "ShareAction", "com/urbanairship/actions/ShareAction.html", null, "" ] ]
, "" ], [ "Enums", null, [ [ "ActionResult.Status", "com/urbanairship/actions/ActionResult.Status.html", null, "" ], [ "Situation", "com/urbanairship/actions/Situation.html", null, "" ] ]
, "" ] ]
, "" ], [ "com.urbanairship.actions.tags", "com/urbanairship/actions/tags/package-summary.html", [ [ "Classes", null, [ [ "AddTagsAction", "com/urbanairship/actions/tags/AddTagsAction.html", null, "" ], [ "BaseTagsAction", "com/urbanairship/actions/tags/BaseTagsAction.html", null, "" ], [ "RemoveTagsAction", "com/urbanairship/actions/tags/RemoveTagsAction.html", null, "" ] ]
, "" ] ]
, "" ], [ "com.urbanairship.amazon", "com/urbanairship/amazon/package-summary.html", [ [ "Classes", null, [ [ "ADMUtils", "com/urbanairship/amazon/ADMUtils.html", null, "" ] ]
, "" ] ]
, "" ], [ "com.urbanairship.analytics", "com/urbanairship/analytics/package-summary.html", [ [ "Classes", null, [ [ "Analytics", "com/urbanairship/analytics/Analytics.html", null, "" ], [ "CustomEvent", "com/urbanairship/analytics/CustomEvent.html", null, "" ], [ "CustomEvent.Builder", "com/urbanairship/analytics/CustomEvent.Builder.html", null, "" ], [ "DefaultEnvironment", "com/urbanairship/analytics/DefaultEnvironment.html", null, "" ], [ "Environment", "com/urbanairship/analytics/Environment.html", null, "" ], [ "Event", "com/urbanairship/analytics/Event.html", null, "" ], [ "EventService", "com/urbanairship/analytics/EventService.html", null, "" ], [ "InstrumentedActivity", "com/urbanairship/analytics/InstrumentedActivity.html", null, "" ], [ "InstrumentedListActivity", "com/urbanairship/analytics/InstrumentedListActivity.html", null, "" ], [ "LocationEvent", "com/urbanairship/analytics/LocationEvent.html", null, "" ] ]
, "" ], [ "Enums", null, [ [ "LocationEvent.UpdateType", "com/urbanairship/analytics/LocationEvent.UpdateType.html", null, "" ] ]
, "" ] ]
, "" ], [ "com.urbanairship.google", "com/urbanairship/google/package-summary.html", [ [ "Classes", null, [ [ "GCMUtils", "com/urbanairship/google/GCMUtils.html", null, "" ], [ "PlayServicesErrorActivity", "com/urbanairship/google/PlayServicesErrorActivity.html", null, "" ], [ "PlayServicesErrorActivity.ErrorDialogFragment", "com/urbanairship/google/PlayServicesErrorActivity.ErrorDialogFragment.html", null, "" ], [ "PlayServicesUtils", "com/urbanairship/google/PlayServicesUtils.html", null, "" ] ]
, "" ] ]
, "" ], [ "com.urbanairship.js", "com/urbanairship/js/package-summary.html", [ [ "Classes", null, [ [ "NativeBridge", "com/urbanairship/js/NativeBridge.html", null, "" ], [ "UAJavascriptInterface", "com/urbanairship/js/UAJavascriptInterface.html", null, "" ] ]
, "" ] ]
, "" ], [ "com.urbanairship.location", "com/urbanairship/location/package-summary.html", [ [ "Interfaces", null, [ [ "LocationListener", "com/urbanairship/location/LocationListener.html", null, "" ] ]
, "" ], [ "Classes", null, [ [ "LocationRequestOptions", "com/urbanairship/location/LocationRequestOptions.html", null, "" ], [ "LocationRequestOptions.Builder", "com/urbanairship/location/LocationRequestOptions.Builder.html", null, "" ], [ "LocationService", "com/urbanairship/location/LocationService.html", null, "" ], [ "UALocationManager", "com/urbanairship/location/UALocationManager.html", null, "" ] ]
, "" ] ]
, "" ], [ "com.urbanairship.preference", "com/urbanairship/preference/package-summary.html", [ [ "Interfaces", null, [ [ "UAPreference", "com/urbanairship/preference/UAPreference.html", null, "" ] ]
, "" ], [ "Classes", null, [ [ "ChannelIdPreference", "com/urbanairship/preference/ChannelIdPreference.html", null, "" ], [ "LocationBackgroundUpdatesAllowedPreference", "com/urbanairship/preference/LocationBackgroundUpdatesAllowedPreference.html", null, "" ], [ "LocationUpdatesEnabledPreference", "com/urbanairship/preference/LocationUpdatesEnabledPreference.html", null, "" ], [ "PushEnablePreference", "com/urbanairship/preference/PushEnablePreference.html", null, "" ], [ "QuietTimeEnablePreference", "com/urbanairship/preference/QuietTimeEnablePreference.html", null, "" ], [ "QuietTimeEndPreference", "com/urbanairship/preference/QuietTimeEndPreference.html", null, "" ], [ "QuietTimeStartPreference", "com/urbanairship/preference/QuietTimeStartPreference.html", null, "" ], [ "SoundEnablePreference", "com/urbanairship/preference/SoundEnablePreference.html", null, "" ], [ "UAPreferenceAdapter", "com/urbanairship/preference/UAPreferenceAdapter.html", null, "" ], [ "UserIdPreference", "com/urbanairship/preference/UserIdPreference.html", null, "" ], [ "VibrateEnablePreference", "com/urbanairship/preference/VibrateEnablePreference.html", null, "" ] ]
, "" ], [ "Enums", null, [ [ "UAPreference.PreferenceType", "com/urbanairship/preference/UAPreference.PreferenceType.html", null, "" ] ]
, "" ] ]
, "" ], [ "com.urbanairship.push", "com/urbanairship/push/package-summary.html", [ [ "Interfaces", null, [ [ "GCMConstants", "com/urbanairship/push/GCMConstants.html", null, "" ] ]
, "" ], [ "Classes", null, [ [ "ADMPushReceiver", "com/urbanairship/push/ADMPushReceiver.html", null, "" ], [ "BaseIntentReceiver", "com/urbanairship/push/BaseIntentReceiver.html", null, "" ], [ "GCMPushReceiver", "com/urbanairship/push/GCMPushReceiver.html", null, "" ], [ "PushManager", "com/urbanairship/push/PushManager.html", null, "" ], [ "PushMessage", "com/urbanairship/push/PushMessage.html", null, "" ], [ "PushService", "com/urbanairship/push/PushService.html", null, "" ] ]
, "" ] ]
, "" ], [ "com.urbanairship.push.notifications", "com/urbanairship/push/notifications/package-summary.html", [ [ "Classes", null, [ [ "CustomLayoutNotificationFactory", "com/urbanairship/push/notifications/CustomLayoutNotificationFactory.html", null, "" ], [ "DefaultNotificationFactory", "com/urbanairship/push/notifications/DefaultNotificationFactory.html", null, "" ], [ "LocalizableRemoteInput", "com/urbanairship/push/notifications/LocalizableRemoteInput.html", null, "" ], [ "LocalizableRemoteInput.Builder", "com/urbanairship/push/notifications/LocalizableRemoteInput.Builder.html", null, "" ], [ "NotificationActionButton", "com/urbanairship/push/notifications/NotificationActionButton.html", null, "" ], [ "NotificationActionButton.Builder", "com/urbanairship/push/notifications/NotificationActionButton.Builder.html", null, "" ], [ "NotificationActionButtonGroup", "com/urbanairship/push/notifications/NotificationActionButtonGroup.html", null, "" ], [ "NotificationActionButtonGroup.Builder", "com/urbanairship/push/notifications/NotificationActionButtonGroup.Builder.html", null, "" ], [ "NotificationFactory", "com/urbanairship/push/notifications/NotificationFactory.html", null, "" ], [ "SystemNotificationFactory", "com/urbanairship/push/notifications/SystemNotificationFactory.html", null, "" ] ]
, "" ] ]
, "" ], [ "com.urbanairship.richpush", "com/urbanairship/richpush/package-summary.html", [ [ "Interfaces", null, [ [ "RichPushInbox.Listener", "com/urbanairship/richpush/RichPushInbox.Listener.html", null, "" ], [ "RichPushManager.Listener", "com/urbanairship/richpush/RichPushManager.Listener.html", null, "" ], [ "RichPushManager.RefreshMessagesCallback", "com/urbanairship/richpush/RichPushManager.RefreshMessagesCallback.html", null, "" ] ]
, "" ], [ "Classes", null, [ [ "RichPushInbox", "com/urbanairship/richpush/RichPushInbox.html", null, "" ], [ "RichPushManager", "com/urbanairship/richpush/RichPushManager.html", null, "" ], [ "RichPushMessage", "com/urbanairship/richpush/RichPushMessage.html", null, "" ], [ "RichPushUpdateService", "com/urbanairship/richpush/RichPushUpdateService.html", null, "" ], [ "RichPushUser", "com/urbanairship/richpush/RichPushUser.html", null, "" ] ]
, "" ] ]
, "" ], [ "com.urbanairship.widget", "com/urbanairship/widget/package-summary.html", [ [ "Classes", null, [ [ "LandingPageWebView", "com/urbanairship/widget/LandingPageWebView.html", null, "" ], [ "RichPushMessageWebView", "com/urbanairship/widget/RichPushMessageWebView.html", null, "" ], [ "UAWebView", "com/urbanairship/widget/UAWebView.html", null, "" ], [ "UAWebViewClient", "com/urbanairship/widget/UAWebViewClient.html", null, "" ] ]
, "" ] ]
, "" ] ]

;



javadocs/reference/lists.js
var DATA = [
      { id:0, label:"com.urbanairship", link:"com/urbanairship/package-summary.html", type:"package" },
      { id:1, label:"com.urbanairship.AirshipConfigOptions", link:"com/urbanairship/AirshipConfigOptions.html", type:"class" },
      { id:2, label:"com.urbanairship.ApplicationMetrics", link:"com/urbanairship/ApplicationMetrics.html", type:"class" },
      { id:3, label:"com.urbanairship.Autopilot", link:"com/urbanairship/Autopilot.html", type:"class" },
      { id:4, label:"com.urbanairship.BaseManager", link:"com/urbanairship/BaseManager.html", type:"class" },
      { id:5, label:"com.urbanairship.Cancelable", link:"com/urbanairship/Cancelable.html", type:"class" },
      { id:6, label:"com.urbanairship.ConstantClass", link:"com/urbanairship/ConstantClass.html", type:"class" },
      { id:7, label:"com.urbanairship.CoreActivity", link:"com/urbanairship/CoreActivity.html", type:"class" },
      { id:8, label:"com.urbanairship.CoreReceiver", link:"com/urbanairship/CoreReceiver.html", type:"class" },
      { id:9, label:"com.urbanairship.Logger", link:"com/urbanairship/Logger.html", type:"class" },
      { id:10, label:"com.urbanairship.PendingResult", link:"com/urbanairship/PendingResult.html", type:"class" },
      { id:11, label:"com.urbanairship.PendingResult.ResultCallback", link:"com/urbanairship/PendingResult.ResultCallback.html", type:"class" },
      { id:12, label:"com.urbanairship.PropertyName", link:"com/urbanairship/PropertyName.html", type:"class" },
      { id:13, label:"com.urbanairship.UAirship", link:"com/urbanairship/UAirship.html", type:"class" },
      { id:14, label:"com.urbanairship.UAirship.OnReadyCallback", link:"com/urbanairship/UAirship.OnReadyCallback.html", type:"class" },
      { id:15, label:"com.urbanairship.UrbanAirshipProvider", link:"com/urbanairship/UrbanAirshipProvider.html", type:"class" },
      { id:16, label:"com.urbanairship.UrbanAirshipResolver", link:"com/urbanairship/UrbanAirshipResolver.html", type:"class" },
      { id:17, label:"com.urbanairship.actions", link:"com/urbanairship/actions/package-summary.html", type:"package" },
      { id:18, label:"com.urbanairship.actions.Action", link:"com/urbanairship/actions/Action.html", type:"class" },
      { id:19, label:"com.urbanairship.actions.Action.ActivityResult", link:"com/urbanairship/actions/Action.ActivityResult.html", type:"class" },
      { id:20, label:"com.urbanairship.actions.ActionActivity", link:"com/urbanairship/actions/ActionActivity.html", type:"class" },
      { id:21, label:"com.urbanairship.actions.ActionArguments", link:"com/urbanairship/actions/ActionArguments.html", type:"class" },
      { id:22, label:"com.urbanairship.actions.ActionArguments.Builder", link:"com/urbanairship/actions/ActionArguments.Builder.html", type:"class" },
      { id:23, label:"com.urbanairship.actions.ActionCompletionCallback", link:"com/urbanairship/actions/ActionCompletionCallback.html", type:"class" },
      { id:24, label:"com.urbanairship.actions.ActionRegistry", link:"com/urbanairship/actions/ActionRegistry.html", type:"class" },
      { id:25, label:"com.urbanairship.actions.ActionRegistry.Entry", link:"com/urbanairship/actions/ActionRegistry.Entry.html", type:"class" },
      { id:26, label:"com.urbanairship.actions.ActionResult", link:"com/urbanairship/actions/ActionResult.html", type:"class" },
      { id:27, label:"com.urbanairship.actions.ActionResult.Status", link:"com/urbanairship/actions/ActionResult.Status.html", type:"class" },
      { id:28, label:"com.urbanairship.actions.ActionRunner", link:"com/urbanairship/actions/ActionRunner.html", type:"class" },
      { id:29, label:"com.urbanairship.actions.ActionService", link:"com/urbanairship/actions/ActionService.html", type:"class" },
      { id:30, label:"com.urbanairship.actions.ActionUtils", link:"com/urbanairship/actions/ActionUtils.html", type:"class" },
      { id:31, label:"com.urbanairship.actions.AddCustomEventAction", link:"com/urbanairship/actions/AddCustomEventAction.html", type:"class" },
      { id:32, label:"com.urbanairship.actions.DeepLinkAction", link:"com/urbanairship/actions/DeepLinkAction.html", type:"class" },
      { id:33, label:"com.urbanairship.actions.LandingPageAction", link:"com/urbanairship/actions/LandingPageAction.html", type:"class" },
      { id:34, label:"com.urbanairship.actions.LandingPageActivity", link:"com/urbanairship/actions/LandingPageActivity.html", type:"class" },
      { id:35, label:"com.urbanairship.actions.OpenExternalUrlAction", link:"com/urbanairship/actions/OpenExternalUrlAction.html", type:"class" },
      { id:36, label:"com.urbanairship.actions.ShareAction", link:"com/urbanairship/actions/ShareAction.html", type:"class" },
      { id:37, label:"com.urbanairship.actions.Situation", link:"com/urbanairship/actions/Situation.html", type:"class" },
      { id:38, label:"com.urbanairship.actions.tags", link:"com/urbanairship/actions/tags/package-summary.html", type:"package" },
      { id:39, label:"com.urbanairship.actions.tags.AddTagsAction", link:"com/urbanairship/actions/tags/AddTagsAction.html", type:"class" },
      { id:40, label:"com.urbanairship.actions.tags.BaseTagsAction", link:"com/urbanairship/actions/tags/BaseTagsAction.html", type:"class" },
      { id:41, label:"com.urbanairship.actions.tags.RemoveTagsAction", link:"com/urbanairship/actions/tags/RemoveTagsAction.html", type:"class" },
      { id:42, label:"com.urbanairship.amazon", link:"com/urbanairship/amazon/package-summary.html", type:"package" },
      { id:43, label:"com.urbanairship.amazon.ADMUtils", link:"com/urbanairship/amazon/ADMUtils.html", type:"class" },
      { id:44, label:"com.urbanairship.analytics", link:"com/urbanairship/analytics/package-summary.html", type:"package" },
      { id:45, label:"com.urbanairship.analytics.Analytics", link:"com/urbanairship/analytics/Analytics.html", type:"class" },
      { id:46, label:"com.urbanairship.analytics.CustomEvent", link:"com/urbanairship/analytics/CustomEvent.html", type:"class" },
      { id:47, label:"com.urbanairship.analytics.CustomEvent.Builder", link:"com/urbanairship/analytics/CustomEvent.Builder.html", type:"class" },
      { id:48, label:"com.urbanairship.analytics.DefaultEnvironment", link:"com/urbanairship/analytics/DefaultEnvironment.html", type:"class" },
      { id:49, label:"com.urbanairship.analytics.Environment", link:"com/urbanairship/analytics/Environment.html", type:"class" },
      { id:50, label:"com.urbanairship.analytics.Event", link:"com/urbanairship/analytics/Event.html", type:"class" },
      { id:51, label:"com.urbanairship.analytics.EventService", link:"com/urbanairship/analytics/EventService.html", type:"class" },
      { id:52, label:"com.urbanairship.analytics.InstrumentedActivity", link:"com/urbanairship/analytics/InstrumentedActivity.html", type:"class" },
      { id:53, label:"com.urbanairship.analytics.InstrumentedListActivity", link:"com/urbanairship/analytics/InstrumentedListActivity.html", type:"class" },
      { id:54, label:"com.urbanairship.analytics.LocationEvent", link:"com/urbanairship/analytics/LocationEvent.html", type:"class" },
      { id:55, label:"com.urbanairship.analytics.LocationEvent.UpdateType", link:"com/urbanairship/analytics/LocationEvent.UpdateType.html", type:"class" },
      { id:56, label:"com.urbanairship.google", link:"com/urbanairship/google/package-summary.html", type:"package" },
      { id:57, label:"com.urbanairship.google.GCMUtils", link:"com/urbanairship/google/GCMUtils.html", type:"class" },
      { id:58, label:"com.urbanairship.google.PlayServicesErrorActivity", link:"com/urbanairship/google/PlayServicesErrorActivity.html", type:"class" },
      { id:59, label:"com.urbanairship.google.PlayServicesErrorActivity.ErrorDialogFragment", link:"com/urbanairship/google/PlayServicesErrorActivity.ErrorDialogFragment.html", type:"class" },
      { id:60, label:"com.urbanairship.google.PlayServicesUtils", link:"com/urbanairship/google/PlayServicesUtils.html", type:"class" },
      { id:61, label:"com.urbanairship.js", link:"com/urbanairship/js/package-summary.html", type:"package" },
      { id:62, label:"com.urbanairship.js.NativeBridge", link:"com/urbanairship/js/NativeBridge.html", type:"class" },
      { id:63, label:"com.urbanairship.js.UAJavascriptInterface", link:"com/urbanairship/js/UAJavascriptInterface.html", type:"class" },
      { id:64, label:"com.urbanairship.location", link:"com/urbanairship/location/package-summary.html", type:"package" },
      { id:65, label:"com.urbanairship.location.LocationListener", link:"com/urbanairship/location/LocationListener.html", type:"class" },
      { id:66, label:"com.urbanairship.location.LocationRequestOptions", link:"com/urbanairship/location/LocationRequestOptions.html", type:"class" },
      { id:67, label:"com.urbanairship.location.LocationRequestOptions.Builder", link:"com/urbanairship/location/LocationRequestOptions.Builder.html", type:"class" },
      { id:68, label:"com.urbanairship.location.LocationService", link:"com/urbanairship/location/LocationService.html", type:"class" },
      { id:69, label:"com.urbanairship.location.UALocationManager", link:"com/urbanairship/location/UALocationManager.html", type:"class" },
      { id:70, label:"com.urbanairship.preference", link:"com/urbanairship/preference/package-summary.html", type:"package" },
      { id:71, label:"com.urbanairship.preference.ChannelIdPreference", link:"com/urbanairship/preference/ChannelIdPreference.html", type:"class" },
      { id:72, label:"com.urbanairship.preference.LocationBackgroundUpdatesAllowedPreference", link:"com/urbanairship/preference/LocationBackgroundUpdatesAllowedPreference.html", type:"class" },
      { id:73, label:"com.urbanairship.preference.LocationUpdatesEnabledPreference", link:"com/urbanairship/preference/LocationUpdatesEnabledPreference.html", type:"class" },
      { id:74, label:"com.urbanairship.preference.PushEnablePreference", link:"com/urbanairship/preference/PushEnablePreference.html", type:"class" },
      { id:75, label:"com.urbanairship.preference.QuietTimeEnablePreference", link:"com/urbanairship/preference/QuietTimeEnablePreference.html", type:"class" },
      { id:76, label:"com.urbanairship.preference.QuietTimeEndPreference", link:"com/urbanairship/preference/QuietTimeEndPreference.html", type:"class" },
      { id:77, label:"com.urbanairship.preference.QuietTimeStartPreference", link:"com/urbanairship/preference/QuietTimeStartPreference.html", type:"class" },
      { id:78, label:"com.urbanairship.preference.SoundEnablePreference", link:"com/urbanairship/preference/SoundEnablePreference.html", type:"class" },
      { id:79, label:"com.urbanairship.preference.UAPreference", link:"com/urbanairship/preference/UAPreference.html", type:"class" },
      { id:80, label:"com.urbanairship.preference.UAPreference.PreferenceType", link:"com/urbanairship/preference/UAPreference.PreferenceType.html", type:"class" },
      { id:81, label:"com.urbanairship.preference.UAPreferenceAdapter", link:"com/urbanairship/preference/UAPreferenceAdapter.html", type:"class" },
      { id:82, label:"com.urbanairship.preference.UserIdPreference", link:"com/urbanairship/preference/UserIdPreference.html", type:"class" },
      { id:83, label:"com.urbanairship.preference.VibrateEnablePreference", link:"com/urbanairship/preference/VibrateEnablePreference.html", type:"class" },
      { id:84, label:"com.urbanairship.push", link:"com/urbanairship/push/package-summary.html", type:"package" },
      { id:85, label:"com.urbanairship.push.ADMPushReceiver", link:"com/urbanairship/push/ADMPushReceiver.html", type:"class" },
      { id:86, label:"com.urbanairship.push.BaseIntentReceiver", link:"com/urbanairship/push/BaseIntentReceiver.html", type:"class" },
      { id:87, label:"com.urbanairship.push.GCMConstants", link:"com/urbanairship/push/GCMConstants.html", type:"class" },
      { id:88, label:"com.urbanairship.push.GCMPushReceiver", link:"com/urbanairship/push/GCMPushReceiver.html", type:"class" },
      { id:89, label:"com.urbanairship.push.PushManager", link:"com/urbanairship/push/PushManager.html", type:"class" },
      { id:90, label:"com.urbanairship.push.PushMessage", link:"com/urbanairship/push/PushMessage.html", type:"class" },
      { id:91, label:"com.urbanairship.push.PushService", link:"com/urbanairship/push/PushService.html", type:"class" },
      { id:92, label:"com.urbanairship.push.notifications", link:"com/urbanairship/push/notifications/package-summary.html", type:"package" },
      { id:93, label:"com.urbanairship.push.notifications.CustomLayoutNotificationFactory", link:"com/urbanairship/push/notifications/CustomLayoutNotificationFactory.html", type:"class" },
      { id:94, label:"com.urbanairship.push.notifications.DefaultNotificationFactory", link:"com/urbanairship/push/notifications/DefaultNotificationFactory.html", type:"class" },
      { id:95, label:"com.urbanairship.push.notifications.LocalizableRemoteInput", link:"com/urbanairship/push/notifications/LocalizableRemoteInput.html", type:"class" },
      { id:96, label:"com.urbanairship.push.notifications.LocalizableRemoteInput.Builder", link:"com/urbanairship/push/notifications/LocalizableRemoteInput.Builder.html", type:"class" },
      { id:97, label:"com.urbanairship.push.notifications.NotificationActionButton", link:"com/urbanairship/push/notifications/NotificationActionButton.html", type:"class" },
      { id:98, label:"com.urbanairship.push.notifications.NotificationActionButton.Builder", link:"com/urbanairship/push/notifications/NotificationActionButton.Builder.html", type:"class" },
      { id:99, label:"com.urbanairship.push.notifications.NotificationActionButtonGroup", link:"com/urbanairship/push/notifications/NotificationActionButtonGroup.html", type:"class" },
      { id:100, label:"com.urbanairship.push.notifications.NotificationActionButtonGroup.Builder", link:"com/urbanairship/push/notifications/NotificationActionButtonGroup.Builder.html", type:"class" },
      { id:101, label:"com.urbanairship.push.notifications.NotificationFactory", link:"com/urbanairship/push/notifications/NotificationFactory.html", type:"class" },
      { id:102, label:"com.urbanairship.push.notifications.SystemNotificationFactory", link:"com/urbanairship/push/notifications/SystemNotificationFactory.html", type:"class" },
      { id:103, label:"com.urbanairship.richpush", link:"com/urbanairship/richpush/package-summary.html", type:"package" },
      { id:104, label:"com.urbanairship.richpush.RichPushInbox", link:"com/urbanairship/richpush/RichPushInbox.html", type:"class" },
      { id:105, label:"com.urbanairship.richpush.RichPushInbox.Listener", link:"com/urbanairship/richpush/RichPushInbox.Listener.html", type:"class" },
      { id:106, label:"com.urbanairship.richpush.RichPushManager", link:"com/urbanairship/richpush/RichPushManager.html", type:"class" },
      { id:107, label:"com.urbanairship.richpush.RichPushManager.Listener", link:"com/urbanairship/richpush/RichPushManager.Listener.html", type:"class" },
      { id:108, label:"com.urbanairship.richpush.RichPushManager.RefreshMessagesCallback", link:"com/urbanairship/richpush/RichPushManager.RefreshMessagesCallback.html", type:"class" },
      { id:109, label:"com.urbanairship.richpush.RichPushMessage", link:"com/urbanairship/richpush/RichPushMessage.html", type:"class" },
      { id:110, label:"com.urbanairship.richpush.RichPushUpdateService", link:"com/urbanairship/richpush/RichPushUpdateService.html", type:"class" },
      { id:111, label:"com.urbanairship.richpush.RichPushUser", link:"com/urbanairship/richpush/RichPushUser.html", type:"class" },
      { id:112, label:"com.urbanairship.widget", link:"com/urbanairship/widget/package-summary.html", type:"package" },
      { id:113, label:"com.urbanairship.widget.LandingPageWebView", link:"com/urbanairship/widget/LandingPageWebView.html", type:"class" },
      { id:114, label:"com.urbanairship.widget.RichPushMessageWebView", link:"com/urbanairship/widget/RichPushMessageWebView.html", type:"class" },
      { id:115, label:"com.urbanairship.widget.UAWebView", link:"com/urbanairship/widget/UAWebView.html", type:"class" },
      { id:116, label:"com.urbanairship.widget.UAWebViewClient", link:"com/urbanairship/widget/UAWebViewClient.html", type:"class" }

    ];
